XiEditor/AppDelegate.swift

```diff
@@ -121,6 +121,18 @@ class AppDelegate: NSObject, NSApplicationDelegate {
                 guard let doc = doc as? Document else { continue }
                 doc.editViewController?.themeChanged(name)
             }
+        case "update_cmds":
+            guard let obj = params as? [String : AnyObject],
+                let view_id = obj["view_id"] as? String,
+                let document = documentForViewIdentifier(viewIdentifier: view_id),
+                let cmds = obj["cmds"] as? [[String: AnyObject]],
+                let plugin = obj["plugin"] as? String else { print("missing plugin field in \(params)"); return nil }
+            let parsedCommands = cmds.map { Command(jsonObject: $0) }
+                .filter { $0 != nil }
+                .map { $0! }
```

Comment:
the `filter` and `map` after the first `map` can be eliminated by turning the first `map` into a `flatMap`.